packages/generator-single-spa/src/react/generator-single-spa-react.js

```diff
@@ -105,18 +122,18 @@ module.exports = class SingleSpaReactGenerator extends Generator {
     );
     this.fs.copyTpl(
       this.templatePath("src/root.component.js"),
-      this.destinationPath("src/root.component.js"),
+      this.destinationPath(`src/root.component.${srcFileExtension}`),
```

Comment:
typescript react files should have `.tsx` file extensions

Reply:
Dumb question is that required or just community norms?
`.ts` = .typescript
`.jsx` for file contains jsx
`.tsx` for file is typescript and contains jsx.

IIRC It's not a hard requirement to use `.jsx` with webpack is it a hard requirement to use `.tsx` with typescript compiler?

Reply:
It is a hard requirement of typescript. You absolutely MUST use `.ts` for typescript files, and MUST use `.tsx` if it has JSX in it.

Reply:
I had no idea it was a hard requirement. Good to know. Thank you.